### PR TITLE
CORE-9556 Add StarkNet Support

### PR DESCRIPTION
**Pull request by erwin-wee**
## Description
- refactor `test_data\evm.py` and a little to enable `test_data\starknet.py` to inherit from `EVMTestData` class and reduce code duplication
- add `url_postfix` argument to `_post` and `make_call` in user\base.py
- add starknet wallet profile
- add `StarkNetTestData` and `StarkNetBenchUser` classes

### CODE REVIEW

**Review by NikitosnikN — APPROVED** [review]
LGTM

File: chainbench/test_data/starknet.py [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+        1536727068981429685321: {
+            "name": "starknet-testnet",
+            "start_block": 500000,
+            "end_block": 890000,
+        },
+        393402129659245999442226: {
+            "name": "starknet-testnet2",
+            "start_block": 1,
+            "end_block": 149000,
+        },
```
Comment (denis-at-chainstack) [inline chainbench/test_data/starknet.py:26]: testnet2 is deprecated

File: chainbench/profile/starknet/wallet.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    def simulate_transaction_task(self):
+        self.make_call(
+            name="simulate_transaction",
+            method="starknet_simulateTransaction",
+            params=self._simulate_transaction_params_factory(get_rng()),
+            url_postfix="/rpc/v0.3",
+        ),
+
+    @task(1)
+    def estimate_fee_task(self):
```
Comment (denis-at-chainstack) [inline chainbench/profile/starknet/wallet.py:77]: will it work for both mainnet and testnet?

File: chainbench/user/starknet.py [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+            "request": {
+                "contract_address": "0x10884171baf1914edc28d7afb619b40a4051cfae78a094a55d230f19e944a28",
+                "entry_point_selector": "0x279193ae67f7ef3a6be330f5bd004266a0ec3fd5a6f7d2fe71a2096b3101578",
+                "calldata": [hex(rng.random.randint(1, 8))],
+            },
+            "block_id": "pending",
+        }
+
+    @staticmethod
+    def _simulate_transaction_params_factory(rng: RNG):
```
Comment (denis-at-chainstack) [inline chainbench/user/starknet.py:22]: will it work for both mainnet and testnet?

**Review by denis-at-chainstack — APPROVED** [review]

**Pull request merged**